tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -24,15 +27,54 @@ internal class Security : IDatadogSecurity, IDisposable
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Security>();
 
+        private static readonly Dictionary<string, string> RequestHeaders;
+        private static readonly Dictionary<string, string> ResponseHeaders;
+
         private static Security _instance;
         private static bool _globalInstanceInitialized;
         private static object _globalInstanceLock = new();
 
         private readonly IWaf _waf;
-        private readonly IAppSecAgentWriter _agentWriter;
         private readonly InstrumentationGateway _instrumentationGateway;
         private readonly SecuritySettings _settings;
 
+        static Security()
+        {
+            var requestHeaders = new List<string>()
+            {
+                "X-FORWARDED-FOR",
+                "X-CLIENT-IP",
+                "X-REAL-IP",
+                "X-FORWARDED",
+                "X-CLUSTER-CLIENT-IP",
+                "FORWARDED-FOR",
+                "FORWARDED",
+                "VIA",
+                "TRUE-CLIENT-IP",
+                "Content-Length",
+                "Content-Type",
+                "Content-Encoding",
+                "Content-Language",
+                "Host",
+                "user-agent",
+                "Accept",
+                "Accept-Encoding",
+                "Accept-Language",
+            };
+
+            RequestHeaders = requestHeaders.ToDictionary(x => x, _ => string.Empty);
+
+            var responseHeaders = new List<string>()
+            {
+                "content-length",
+                "content-type",
+                "Content-Encoding",
+                "Content-Language",
+            };
+
+            ResponseHeaders = responseHeaders.ToDictionary(x => x, _ => string.Empty);
```

Comment:
[nit] Linq and throw-away list seems a bit unnecessary to me
```suggestion
            RequestHeaders = new Dictionary<string, string>()
            {
                { "X-FORWARDED-FOR", string.Empty },
                { "X-CLIENT-IP", string.Empty },
                { "X-REAL-IP", string.Empty },
                { "X-FORWARDED", string.Empty },
                { "X-CLUSTER-CLIENT-IP", string.Empty },
                { "FORWARDED-FOR", string.Empty },
                { "FORWARDED", string.Empty },
                { "VIA", string.Empty },
                { "TRUE-CLIENT-IP", string.Empty },
                { "Content-Length", string.Empty },
                { "Content-Type", string.Empty },
                { "Content-Encoding", string.Empty },
                { "Content-Language", string.Empty },
                { "Host", string.Empty },
                { "user-agent", string.Empty },
                { "Accept", string.Empty },
                { "Accept-Encoding", string.Empty },
                { "Accept-Language", string.Empty },
            };

            ResponseHeaders = new Dictionary<string, string>()
            {
                { "content-length", string.Empty },
                { "content-type", string.Empty },
                { "Content-Encoding", string.Empty },
                { "Content-Language", string.Empty },
            };
```

Reply:
As it happens once at start up, I thought the connivence of just having to modify a list would be worth extra overhead, but as you've made the changes for me, I can't really argue.